code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -213,6 +213,10 @@
 	name = "Unholy Water"
 	id = "unholywater"
 	description = "Something that shouldn't exist on this plane of existence."
+	
+/datum/reagent/fuel/unholywater/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
```

Comment:
you should call the parent proc with ..() at the bottom, so reactions that aren't touch or vapour can be handled by higher parts of the stack

ideally

      return ..()